bin/pytorch_inference/Main.cc

```diff
@@ -274,29 +277,41 @@ int main(int argc, char** argv) {
 
     ml::torch::CCommandParser commandParser{ioMgr.inputStream(), cacheMemorylimitBytes};
 
-    // Size the threadpool to the number of hardware threads
-    // so we can grow and shrink the threadpool dynamically.
-    // The task queue size is set to 1.
-    ml::core::startDefaultAsyncExecutor(0, 1);
-    // Set the number of threads to use
-    ml::core::defaultAsyncExecutor().numberThreadsInUse(threadSettings.numAllocations());
+    if (useImmediateExecutor == false) {
+        // Size the threadpool to the number of hardware threads
+        // so we can grow and shrink the threadpool dynamically.
+        // The task queue size is set to 1.
+        ml::core::startDefaultAsyncExecutor(0, 1);
+        // Set the number of threads to use
+        ml::core::defaultAsyncExecutor().numberThreadsInUse(threadSettings.numAllocations());
+    }
+    ml::core::CExecutor& executor{ml::core::defaultAsyncExecutor()};
+    ml::core::CImmediateExecutor immediateExecutor{ml::core::CImmediateExecutor()};
+
+    if (useImmediateExecutor) {
+        executor = immediateExecutor;
+    }
```

Comment:
If you want immediate execution this is the default if you don't start the async executor or to be explicit you could explicitly stop.
```suggestion
    if (useImmediateExecutor == false) {
        // Size the threadpool to the number of hardware threads
        // so we can grow and shrink the threadpool dynamically.
        // The task queue size is set to 1.
        ml::core::startDefaultAsyncExecutor(0, 1);
        // Set the number of threads to use
        ml::core::defaultAsyncExecutor().numberThreadsInUse(threadSettings.numAllocations());
    } else {
        // Make sure we're using immediate execution.
        ml::core::stopDefaultAsyncExecutor();
    }
```